src/azul/project/hca/transformers.py

```diff
@@ -369,16 +419,30 @@ def transform(self,
                             manifest=manifest,
                             metadata_files=metadata_files)
         project = self._get_project(bundle)
-        for specimen in bundle.specimens:
+        samples: MutableMapping[str, Sample] = dict()
+        for si in bundle.sequencing_input:
+            self.get_ancestor_samples(samples, si)
+        for sample in samples.values():
             visitor = TransformerVisitor()
-            specimen.accept(visitor)
-            specimen.ancestors(visitor)
-            contents = dict(specimens=[_specimen_dict(specimen, visitor.visited_entities)],
+            sample.accept(visitor)
+            sample.ancestors(visitor)
+            contents = dict(samples=[{'sample_type': api.schema_names[type(sample)],
+                                      'document_id': str(sample.document_id),
+                                      'biomaterial_id': str(sample.biomaterial_id)}],
+                            specimens=[_specimen_dict(s) for s in visitor.specimens.values()],
                             cell_suspensions=[_cell_suspension_dict(cs) for cs in visitor.cell_suspensions.values()],
+                            cell_lines=[_cell_line_dict(cl) for cl in visitor.cell_lines.values()],
+                            donor_organisms=[_donor_organism_dict(d) for d in visitor.donor_organisms.values()],
+                            organoids=[_organoid_dict(o) for o in visitor.organoids.values()],
                             files=[_file_dict(f) for f in visitor.files.values()],
                             protocols=[_protocol_dict(pl) for pl in visitor.protocols.values()],
                             projects=[_project_dict(project)])
-            yield self._contribution(bundle, contents, specimen)
+            sample_type = contents['samples'][0]['sample_type']
+            sample_types = {'cell_line': 'cell_lines', 'organoid': 'organoids', 'specimen_from_organism': 'specimens'}
+            assert sample_type in sample_types.keys()
```

Comment:
This is a bit redundant, because the next line would fail with a KeyError, but ok. `k in d.keys()` is the same as `k in d`.